google-cloud-bigquery/src/test/java/com/google/cloud/bigquery/it/ITBigQueryTest.java

```diff
@@ -816,6 +818,77 @@ public void testJsonType() throws InterruptedException {
+      assertNotNull(createdTable);
+
+      // Insert 3 rows of Interval data into the Interval column
+      Map<String, Object> intervalRow1 =
+          Collections.singletonMap("intervalField", "123-7 -19 0:24:12.000006");
+      Map<String, Object> intervalRow2 =
+          Collections.singletonMap("intervalField", "P123Y7M-19DT0H24M12.000006S");
+
+      InsertAllRequest request =
+          InsertAllRequest.newBuilder(tableId).addRow(intervalRow1).addRow(intervalRow2).build();
+      InsertAllResponse response = bigquery.insertAll(request);
+      assertFalse(response.hasErrors());
+      assertEquals(0, response.getInsertErrors().size());
+
+      // Insert another Interval row parsed from a String with Interval positional query parameter
+      String dml = "INSERT INTO " + tableId.getTable() + " (intervalField) VALUES(?)";
+      // Parsing from ISO 8610 format String
+      QueryParameterValue intervalParameter =
+          QueryParameterValue.interval("P125Y7M-19DT0H24M12.000006S");
+      QueryJobConfiguration dmlQueryJobConfiguration =
+          QueryJobConfiguration.newBuilder(dml)
+              .setDefaultDataset(DatasetId.of(DATASET))
+              .setUseLegacySql(false)
+              .addPositionalParameter(intervalParameter)
+              .build();
+      bigquery.query(dmlQueryJobConfiguration);
+      Page<FieldValueList> rows = bigquery.listTableData(tableId);
+      assertEquals(3, Iterables.size(rows.getValues()));
+
+      // Parsing from threeten-extra PeriodDuration
+      QueryParameterValue intervalParameter1 =
+          QueryParameterValue.interval(
+              PeriodDuration.of(Period.of(1, 2, 25), java.time.Duration.ofHours(8)));
+      QueryJobConfiguration dmlQueryJobConfiguration1 =
+          QueryJobConfiguration.newBuilder(dml)
+              .setDefaultDataset(DatasetId.of(DATASET))
+              .setUseLegacySql(false)
+              .addPositionalParameter(intervalParameter1)
+              .build();
+      bigquery.query(dmlQueryJobConfiguration1);
+      Page<FieldValueList> rows1 = bigquery.listTableData(tableId);
+      assertEquals(4, Iterables.size(rows1.getValues()));
+
+      // Query the Interval column with Interval positional query parameter
+      String sql = "SELECT intervalField FROM " + tableId.getTable() + " WHERE intervalField = ? ";
+      QueryParameterValue intervalParameter2 =
+          QueryParameterValue.interval("P125Y7M-19DT0H24M12.000006S");
+      QueryJobConfiguration queryJobConfiguration =
+          QueryJobConfiguration.newBuilder(sql)
+              .setDefaultDataset(DatasetId.of(DATASET))
+              .setUseLegacySql(false)
+              .addPositionalParameter(intervalParameter2)
+              .build();
+      TableResult result = bigquery.query(queryJobConfiguration);
+      for (FieldValueList values : result.iterateAll()) {
+        assertEquals("125-7 -19 0:24:12.000006", values.get(0).getValue());
```

Comment:
Will there be future support to expose conversion to a PeriodDuration, given we accept it on the query parameter inputs?

Reply:
+1 to adding a `getPeriodDuration()` function here: https://cloud.google.com/java/docs/reference/google-cloud-bigquery/latest/com.google.cloud.bigquery.FieldValue

Reply:
Yes, this is a great suggestion. I created a GH issue to keep track of this feature enhancement: https://github.com/googleapis/java-bigquery/issues/1849.